nodejs/eks/cni-addon.ts

```diff
@@ -0,0 +1,545 @@
+// Copyright 2016-2014, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as k8s from "@pulumi/kubernetes";
+import * as deepmerge from "deepmerge";
+
+export interface VpcCniAddonOptions {
+    clusterName: pulumi.Input<string>;
+
+    /**
+     * The Kubernetes version of the cluster. This is used to determine the addon version to use if `addonVersion` is not specified.
+     */
+    clusterVersion?: pulumi.Input<string>;
+
+    /**
+     * How to resolve field value conflicts when migrating a self-managed add-on to an Amazon EKS add-on. Valid values are `NONE` and `OVERWRITE`. For more details see the [CreateAddon](https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateAddon.html) API Docs.
+     */
+    resolveConflictsOnCreate?: pulumi.Input<string>;
```

Comment:
Cory is adding enums for the conflict resolution in this PR: https://github.com/pulumi/pulumi-eks/pull/1357

I'm gonna update those params once that's merged in

Reply:
Tracker for tackling the follow up: https://github.com/pulumi/pulumi-eks/issues/1369